edb/server/protocol/binary.pyx

```diff
@@ -2050,8 +2050,15 @@ cdef class EdgeConnection:
         # for empty tuple it's okay to send zero-length arguments
         # TODO(tailhook) force zero-length arguments for protocol 0.12 after
         # upgrading the bindings
-        if frb_get_len(&in_buf) == 0:
-            recv_args = 0
+        if self.protocol_version >= (0, 12):
+            if frb_get_len(&in_buf) == 0:
```

Comment:
I think `CompiledQuery` must have the input type descriptor -- we need to also check that it's zero type.

Reply:
changed